feature/gcp ci

mlawsonca opened the pull request:
- **Dummy commit**
- **Add GCP client Dockerfiles, add unit test in GHA**

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

mlawsonca committed:
Add GCP client Dockerfiles, add unit test in GHA

Run-GHA: true

Required-githooks: true

Signed-off-by: mlawsonca <mlawsonca@google.com>
Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Run Docker as sudo

Run-GHA: true
Required-githooks: true
Signed-off-by: mlawsonca <mlawsonca@google.com>
Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add missing build file

Required-githooks: true

Signed-off-by: mlawsonca <mlawsonca@google.com>
Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add DAOS_SRC_DIR

Required-githooks: true

Signed-off-by: mlawsonca <mlawsonca@google.com>
Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Dummy commit to find missing file

Required-githooks: true

Signed-off-by: mlawsonca <mlawsonca@google.com>
Signed-off-by: Margaret Lawson <mlawsonca@google.com>

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15433/10/display/redirect

daosbuild1 commented:
Test stage Unit Test bdev with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15433/10/display/redirect

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15433/10/display/redirect

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15433/21/display/redirect

daosbuild1 commented:
Test stage Functional Hardware Medium Verbs Provider completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15433/21/display/redirect

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-15433/21/execution/node/1531/log

mlawsonca committed:
Update get_rpm_hash to use hashes that match utils/build.config

Required-githooks: true

Signed-off-by: mlawsonca <mlawsonca@google.com>
Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Update how we pull Raft

Required-githooks: true

Signed-off-by: mlawsonca <mlawsonca@google.com>
Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Update isal_crypto to 2.24.0, add raft HEAD

isal_crypto 2.23.0 is broken out of the box for el8

Raft module head needed to pull correct version in Dockerfile

Required-githooks: true

Signed-off-by: mlawsonca <mlawsonca@google.com>
Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
ADD in needed daos.spec

Adjust for the fact that we aren't including the entire .git folder

Required-githooks: true

Signed-off-by: mlawsonca <mlawsonca@google.com>
Signed-off-by: Margaret Lawson <mlawsonca@google.com>

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15433/24/display/redirect

daosbuild1 commented:
Test stage Functional Hardware Medium Verbs Provider completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15433/24/display/redirect

mlawsonca committed:
Draft commit - needs to be fixed

Required-githooks: true

Signed-off-by: mlawsonca <mlawsonca@google.com>
Signed-off-by: Margaret Lawson <mlawsonca@google.com>

daosbuild1 commented:
Test stage Functional Hardware Medium Verbs Provider completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15433/25/display/redirect

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15433/25/display/redirect

mlawsonca committed:
Trying to use existing RPM packaging scripts

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Update workflow to trigger run

Required-githooks: true
Run-GHA: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Fix file name

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add temp test

Run-GHA: true

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Revert "Add temp test"

This reverts commit 508f9ccce2d04bcc7808a7b6ec1d12fd16c0c3c5.

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Revert "Fix file name"

This reverts commit d87c8dc46e1725723f16de8fecef5af90da14825.

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Revert "Update workflow to trigger run"

This reverts commit 0e23d5f547b6aa20c5503c69b4b2ebf149cf5af2.

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Revert "Trying to use existing RPM packaging scripts"

This reverts commit 109dfa009526464dcf5640c4ef98b13b0255f386.

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Revert "Draft commit - needs to be fixed"

This reverts commit 19b3d9483d2fd0780be2e9b3d342e1bc9d9162b8.

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add in RPM build test

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Remove google auth step

Run-GHA: true

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add missing env variable

Run-GHA: true

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add Docker run permissions

Run-GHA: true

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Update to use EL8.9 for build

Required-githooks: true
Run-GHA: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Reset EL build version to 8.6

Run-GHA: true

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Reset EL version to 8.9

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add GCP auth test, temporarily disable others

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Try removing quote marks around string to see if GCP Auth will work

Run-GHA: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-15433/48/testReport/

mlawsonca committed:
Adjust RPM build test to use GCP Dockerfiles

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Remove .dockerignore file when packaging RPMs

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Update libfabric hash to use v1.22.0

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15433/53/display/redirect

mlawsonca committed:
Fix RPM build issues

Fix broken path
Add empty .dockerignore to prevent RPM packaging error

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-15433/56/testReport/

mlawsonca committed:
Switch to installing Argobots from source

DAOS 2.7 needs argobots-devel >= 1.2, but no publicly available RPM
exists right now for the version

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Fix Debian 12 client RPM build issues

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Renamed files

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add testing client Dockerfiles

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add files for running unit tests and creating client testing Dockerfiles

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Disable unit testing until we can pull the RPMs from Artifact Registry

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add fix for NLT

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Updated hashes, fix GCP Auth test

Run-GHA: True
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Fix small errors

[no ci]
Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Temporarily disable RPM tests

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Re-enable unit testing and add in GCP AR logic

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Update Gcloud Setup GHA version and fix unbound variable

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Update gcloud auth version

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add in missing GHA GCP Auth workflow permissions and make file executable

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Temp, unfinished, middle of changing symlinks

[no-ci]
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Try testing unit tests and NLT

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add missing GHA

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Import commit message fix

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Fix input usage

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Try again to fix inputs

Run-GHA: true
Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>

mlawsonca committed:
Add server build instructions (untested)

[no-ci]

Required-githooks: true

Signed-off-by: Margaret Lawson <mlawsonca@google.com>
